rust-engine/src/backends/lean.rs

```diff
@@ -228,6 +243,41 @@ set_option linter.unusedVariables false
                 .parens()
                 .group()
                 .nest(INDENT),
+                ExprKind::Resugared(resugared_expr_kind) => match resugared_expr_kind {
+                    ResugaredExprKind::BinOp {
+                        op,
+                        lhs,
+                        rhs,
+                        generic_args: _,
+                        bounds_impls: _,
+                        trait_: _,
+                    } => {
+                        let symbol = if op == &binops::add() {
```

Comment:
Should this be a match? That would probably look a little nicer.

Reply:
Makes sense. But also sounds like a design smell.

Reply:
Yeah, let's get this in as is. But maybe let's look at this together when you made the plan for the interning changes you're planning?